PYME/LMVis/Extras/pointSetGeneration.py

```diff
@@ -134,14 +134,23 @@ def OnConfigure(self, event):
         self.edit_traits()
 
     def OnGenPoints(self, event):
-        self.xp, self.yp, self.zp = self.source.getPoints()
+        import wx
+        try:
+            self.xp, self.yp, self.zp = self.source.getPoints()
+        except KeyError:
+            dlg = wx.MessageDialog(self.visFr, "Could not access image, please configure image source",
```

Comment:
This error messages is incorrect (and potentially confusing) for 2/3 cases - i.e. when the source is not an image.

Reply:
While possibly obscure the KeyError is only ever raised with the image source. Trapping the error earlier in the chain does not work. What is the tidy solution, e.g. create a new 'ImageSourceKeyError' and raise and check for that?

Reply:
There's a chance that it doesn't practically matter if the `ImageSource` is the only one which could throw a `KeyError`

Reply:
That is the case in my reading of the code and thus a `KeyError` directly implies the `ImageSource`.